dspace-server-webapp/src/test/java/org/dspace/app/rest/SubscriptionRestRepositoryIT.java

```diff
@@ -0,0 +1,847 @@
+                             .andExpect(jsonPath("$.name", Matchers.is(publicItem.getName())))
+                             .andExpect(jsonPath("$.withdrawn", Matchers.is(false)))
+                             .andExpect(jsonPath("$.discoverable", Matchers.is(true)))
+                             .andExpect(jsonPath("$.inArchive", Matchers.is(true)))
+                             .andExpect(jsonPath("$.type", Matchers.is("item")));
+    }
+
+    @Test
+    public void linkedDSpaceObjectOfSubscriptionTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+        List<SubscriptionParameter> subscriptionParameterList = new ArrayList<>();
+        SubscriptionParameter subscriptionParameter = new SubscriptionParameter();
+        subscriptionParameter.setName("Parameter");
+        subscriptionParameter.setValue("ValueParameter");
+        subscriptionParameterList.add(subscriptionParameter);
+        Subscription subscription = SubscribeBuilder.subscribeBuilder(context,
+                                    "TestType", publicItem, eperson, subscriptionParameterList).build();
+        context.restoreAuthSystemState();
+
+        String tokenAdmin = getAuthToken(eperson.getEmail(), password);
+        getClient(tokenAdmin).perform(get("/api/core/subscriptions/" + subscription.getID() + "/dSpaceObject"))
+                             .andExpect(status().isOk())
+                             .andExpect(jsonPath("$.uuid", Matchers.is(publicItem.getID().toString())))
+                             .andExpect(jsonPath("$.name", Matchers.is(publicItem.getName())))
+                             .andExpect(jsonPath("$.withdrawn", Matchers.is(false)))
+                             .andExpect(jsonPath("$.discoverable", Matchers.is(true)))
+                             .andExpect(jsonPath("$.inArchive", Matchers.is(true)))
+                             .andExpect(jsonPath("$.type", Matchers.is("item")));
+    }
+
+    @Test
+    public void linkedDSpaceObjectOfSubscriptionUnauthorizedTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+        List<SubscriptionParameter> subscriptionParameterList = new ArrayList<>();
+        SubscriptionParameter subscriptionParameter = new SubscriptionParameter();
+        subscriptionParameter.setName("Parameter");
+        subscriptionParameter.setValue("ValueParameter");
+        subscriptionParameterList.add(subscriptionParameter);
+        Subscription subscription = SubscribeBuilder.subscribeBuilder(context,
+                                    "TestType", publicItem, eperson, subscriptionParameterList).build();
+        context.restoreAuthSystemState();
+
+        getClient().perform(get("/api/core/subscriptions/" + subscription.getID() + "/dSpaceObject"))
+                   .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    public void linkedDSpaceObjectOfSubscriptionForbiddenTest() throws Exception {
```

Comment:
We need a test that verifies you cannot see the object if you've been removed from access.  For example:
* Create a subscription to an object for a normal EPerson
* Now access restrict that same object (so that the EPerson no longer has access)
* Verify that the EPerson CANNOT access the object via this link

Currently, I think this type of test might fail based on my reading of the code in this PR.

Reply:
As said here https://github.com/DSpace/DSpace/pull/8620/files#r1094609559
I expect that the request succeed but the response is empty. We need to be sure that no data disclosure happen but it would be ok with both a 200/201 response than with a 500 as this scenario was not managed in the previous versions as well

Reply:
@abollini & @corrad82-4s  : That sounds reasonable to me.  If you have a subscription to an object, and then are *removed* from accessing that object, then it's OK for the subscription to still exist. But we need tests to prove that the subscription will *never* respond with data (so it'd just give an empty response as you said).  

My main point in my prior reviews was to ensure we have tests to prove that it's not possible to continue to receive updates about a Collection if that Collection is later access restricted from you.   So, @abollini 's approach is fine with me... we just need tests to prove it works that way.

Reply:
We still don't have tests here that prove what happens if an object is access restricted **after** you have a subscription to it.  So, I don't see a test that does this:
* Create a subscription to an object for a normal EPerson
* Now access restrict that same object (so that the subscribed EPerson no longer has access)
* Verify that the EPerson CANNOT access the object via this link & no longer gets subscription emails related to this object.

If I've overlooked this test, let me know, but I don't see it.